modules/internal/CPtr.chpl

```diff
@@ -213,43 +213,38 @@ module CPtr {
     return c_pointer_return(x);
   }
 
+  private extern const CHPL_RT_MD_ARRAY_ELEMENTS:chpl_mem_descInt_t;
+
   /*
     Allocate memory that is filled with zeros. This memory should eventually be
-    freed with c_free.
+    freed with :proc:`c_free`.
 
     :arg eltType: the type of the elements to allocate
     :arg size: the number of elements to allocate
     :returns: a c_ptr(eltType) to allocated memory
     */
   inline proc c_calloc(type eltType, size: integral) : c_ptr(eltType) {
-    var ret:c_ptr(eltType);
-    __primitive("array_alloc", ret, eltType, size);
-    // TODO - one day, this should call calloc instead of
-    // initializing the memory possibly in parallel.
-    init_elts(ret, size, eltType);
-    return ret;
+    return chpl_here_calloc(size, 1, CHPL_RT_MD_ARRAY_ELEMENTS):c_ptr(eltType);
```

Comment:
You think it needs an explicit comment here even though c_ptr is described as being for c interop only?

I'm ok with that, just wanted to make sure